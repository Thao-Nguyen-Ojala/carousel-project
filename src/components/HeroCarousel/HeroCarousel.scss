.hero-container {
  padding: 1rem 0;
  border: 0.5px solid #E4E4E4;
  border-radius: 10px;
  box-shadow: 2px 2px 10px #E4E4E4;
  width: 100%;

  &__carousel {
    height: 17rem;

    &__hero-wrapper {
      display: flex;
      flex-direction: row;
      justify-content: center;
      align-items: center;
    
      &__img-section {
        width: 60%;
    
        &__img{
          width: 50rem !important;
          height: 20rem !important;
        }
      }

      &__text-section{
        text-align: center;
        padding-right: 2rem;
        font-size: 21px;
        font-weight: bold;
      }
    }
  }
  
}



@media screen and (max-width:1100px) {
  .hero-container {
    width: 100%;

    &__carousel {

      &__hero-wrapper {
        position: relative;
        text-align: center;
    
        &__img-section {
          display: block;
          width: 90%;
    
          &__img {
            width: 100% !important;
            height: width !important;
          }
        }
    
        &__text-section {
        width: 90%;
        background-color:rgba(255,255,255, 0.3);
        padding: 1rem;
        position: absolute;
        bottom: 16px;
        text-align: center;
      }
      }
    }
  } 
}
@media screen and (max-width:500px) {
  .hero-container{
    padding: 1rem 0;
  }
}

