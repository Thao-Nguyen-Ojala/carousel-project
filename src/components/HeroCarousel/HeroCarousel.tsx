import './HeroCarousel.scss';
import { Blurhash } from 'react-blurhash';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import { RestaurantSection } from '../../restaurant.module';

const responsive = {
  screen: {
    breakpoint: { max: 9999, min: 0 },
    items: 1,
  },
};

type HeroCarouselPropsType = { restaurantSection: RestaurantSection };

const HeroCarousel = ({ restaurantSection }: HeroCarouselPropsType) => {
  const { restaurants } = restaurantSection;
  return (
    <div className="hero-container">
      <Carousel
        responsive={responsive}
        infinite={true}
        renderButtonGroupOutside={true}
        containerClass="hero-container__carousel"
      >
        {restaurants.map((restaurant, i) => {
          return (
            <div className="hero-container__carousel__hero-wrapper" key={i}>
              <div className="hero-container__carousel__hero-wrapper__img-section">
                <Blurhash
                  className="hero-container__carousel__hero-wrapper__img-section__img"
                  hash={restaurant.blurhash}
                  resolutionX={32}
                  resolutionY={32}
                  punch={1}
                />
              </div>
              <span className="hero-container__carousel__hero-wrapper__text-section">
                {restaurant.name}
              </span>
            </div>
          );
        })}
      </Carousel>
    </div>
  );
};

export default HeroCarousel;
