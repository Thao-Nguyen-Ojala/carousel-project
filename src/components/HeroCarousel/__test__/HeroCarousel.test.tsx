import React from 'react';
import ReactDOM from 'react-dom';
import HeroCarousel from '../HeroCarousel';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';

afterEach(cleanup);

const testingObj = {
  title: 'Popular Restaurants',
  restaurants: [
    {
      blurhash: 'UAN=8k?LS~M:ErJFs%t0MDMWRqo@%BxSV{RX',
      launch_date: '2020-04-20',
      location: [24.938082, 60.17626],
      name: 'Sea Chain',
      online: true,
      popularity: 0.956990414084132,
    },
  ],
};

it('render HeroCarousel without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <HeroCarousel restaurantSection={testingObj}></HeroCarousel>,
    div
  );
});

it('matches HeroCarousel snapshot', () => {
  const testingHeroCarousel = renderer
    .create(<HeroCarousel restaurantSection={testingObj}></HeroCarousel>)
    .toJSON();
  expect(testingHeroCarousel).toMatchSnapshot();
});
