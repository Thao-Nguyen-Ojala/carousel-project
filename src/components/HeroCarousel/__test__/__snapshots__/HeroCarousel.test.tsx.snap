// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches HeroCarousel snapshot 1`] = `
<div
  className="hero-container"
>
  <div
    className="react-multi-carousel-list hero-container__carousel "
  >
    <ul
      className="react-multi-carousel-track "
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseMove={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "overflow": "unset",
          "transform": "translate3d(0px,0,0)",
          "transition": "none",
        }
      }
    />
  </div>
</div>
`;
